bofire/strategies/stepwise/stepwise.py

```diff
@@ -41,40 +43,67 @@
     def __init__(self, data_model: data_model, **kwargs):
         super().__init__(data_model, **kwargs)
-        self.steps = data_model.steps
+        self.stratgies = [_map(s.strategy_data) for s in data_model.steps]
+        self.conditions = [s.condition for s in data_model.steps]
+        self.transforms = [
+            s.transform and transforms.map(s.transform) for s in data_model.steps
+        ]
 
     def has_sufficient_experiments(self) -> bool:
         return True
 
-    def _get_step(self) -> Tuple[int, Step]:  # type: ignore
-        for i, step in enumerate(self.steps):
-            condition = conditions.map(step.condition)
+    def _get_step(self) -> Tuple[Strategy, Optional[Transform]]:
+        """Returns index of the current step, the step itself"""
+        for i, condition in enumerate(self.conditions):
             if condition.evaluate(self.domain, experiments=self.experiments):
-                return i, step
+                return self.stratgies[i], self.transforms[i]
         raise ValueError("No condition could be satisfied.")
 
-    def _ask(self, candidate_count: int) -> pd.DataFrame:
-        # we have to decide here w
-        istep, step = self._get_step()
-        if (step.max_parallelism > 0) and (candidate_count > step.max_parallelism):
-            raise ValueError(
-                f"Maximum number of candidates for step {istep} is {step.max_parallelism}."
-            )
-        # map it
-        strategy = map(step.strategy_data)
+    def _ask(self, candidate_count: Optional[PositiveInt]) -> pd.DataFrame:
+        strategy, transform = self._get_step()
+
+        candidate_count = candidate_count or 1
+
+        # handle a possible transform
+        tf_domain = _apply_tf(self.domain, transform, "domain")
+        transformed_domain = tf_domain or self.domain
+        strategy.domain = transformed_domain
+        tf_exp = _apply_tf(self.experiments, transform, "experiments")
+        transformed_experiments = self.experiments if tf_exp is None else tf_exp
+        tf_cand = _apply_tf(self.candidates, transform, "candidates")
+        transformed_candidates = self.candidates if tf_cand is None else tf_cand
         # tell the experiments
-        if self.num_experiments > 0:
-            strategy.tell(experiments=self.experiments)
+        if transformed_experiments is not None and self.num_experiments > 0:
+            strategy.tell(experiments=transformed_experiments, replace=True)
         # tell pending
-        if self.num_candidates > 0:
-            strategy.set_candidates(self.candidates)
+        if transformed_candidates is not None and len(transformed_candidates) > 0:
+            strategy.set_candidates(transformed_candidates)
         # ask and return
-        return strategy.ask(candidate_count=candidate_count)
+        data_asked_for = strategy.ask(candidate_count=candidate_count)
+        if transform is not None:
+            return transform.untransform_experiments(data_asked_for)
```

Comment:
```suggestion
            return transform.untransform_candidates(data_asked_for)
```

why not call `data_asked_for` `candidates`?